core/src/main/scala/caliban/parsing/Parser.scala

```diff
@@ -192,7 +192,7 @@ object Parser {
     }
 
   private def typeDefinition[_: P]: P[TypeDefinition] =
-    P("type" ~/ name ~ "{" ~ fieldDefinition.rep ~ "}").map(t => TypeDefinition(t._1, t._2.toList))
+    P(("type" | "input") ~/ name ~ "{" ~ fieldDefinition.rep ~ "}").map(t => TypeDefinition(t._1, t._2.toList))
```

Comment:
Actually if you look at the spec, an input type is slightly different from a regular type:
http://spec.graphql.org/June2018/#InputValueDefinition
http://spec.graphql.org/June2018/#FieldDefinition
(no arguments possible, default value added)